OpenRA.Mods.Common/Traits/World/MusicPlaylist.cs

```diff
@@ -164,6 +164,14 @@ public void Play(MusicInfo music, Action onComplete)
 			Game.Sound.PlayMusicThen(music, onComplete);
 		}
 
+		public void SetBackgroundMusic(MusicInfo music)
+		{
+			currentSong = currentBackgroundSong = music;
+			CurrentSongIsBackground = music != null;
+
+			Play();
```

Comment:
This seems a bit unintentional though...
To have to call `Media.Stop()` after `Media.PlayBackgroundMusic`.
I'll just add `playlist.Stop()` to `Media.PlayBackgroundMusic` with an extra bool parameter.